crates/bevy_pbr/src/environment_map/diffuse_convolution.wgsl

```diff
@@ -0,0 +1,72 @@
+// Importance samples (Lambertian distribution) a skybox to produce a diffuse lighting cubemap
+// Based on https://github.com/KhronosGroup/glTF-IBL-Sampler/blob/master/lib/source/shaders/filter.frag
+
+#import bevy_pbr::utils::PI
+
+@group(0) @binding(0) var skybox: texture_cube<f32>;
+#ifdef RG11B10FLOAT
+@group(0) @binding(1) var diffuse_map: texture_storage_2d_array<rg11b10float, write>;
+#else
+@group(0) @binding(1) var diffuse_map: texture_storage_2d_array<rgba16float, write>;
+#endif
+@group(0) @binding(2) var bilinear: sampler;
+
+fn get_dir(u: f32, v: f32, face: u32) -> vec3<f32> {
+    switch face {
+        case 0u: { return vec3(1.0, v, -u); }
+        case 1u: { return vec3(-1.0, v, u); }
+        case 2u: { return vec3(u, 1.0, -v); }
+        case 3u: { return vec3(u, -1.0, v); }
+        case 4u: { return vec3(u, v, 1.0); }
+        default { return vec3(-u, v, -1.0); }
+    }
+}
+
+fn generate_tbn(normal: vec3<f32>) -> mat3x3<f32> {
+    var bitangent = vec3(0.0, 1.0, 0.0);
+
+    let n_dot_up = dot(normal, bitangent);
+    if 1.0 - abs(n_dot_up) <= 0.0000001 {
```

Comment:
`if abs(n_dot_up) >= 1 - EPSILON` might be clearer.

Reply:
Done